### ISSUE TITLE
Unsupported Safari version

### ISSUE THREAD

**Issue by harishnb** [google/shaka-player#2548]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
<s>2.8.10</s> (assume you meant 2.5.10?)

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `master`?**
Yes

**Are you using the demo app or your own custom app?**
Demo using the code mentioned at location:  https://shaka-player-demo.appspot.com/docs/api/tutorial-basic-usage.html

**If custom app, can you reproduce the issue using our demo app?**
na

**What browser and OS are you using?**
webkit and safari 9.1 

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
webkit and safari 9.1 

**What are the manifest and license server URIs?**

**What did you do?**

Provided standard mpd but playback stalls in between. Whereas if we run the MSE demo page (mentioned below) no issue is seen.  It would be helpful if you can have a look at the log and share some insights.

**What did you expect to happen?**
The video playback works smoothly.

**What actually happened?**
Once the video playback starts after 10 sec or so the video stalls, have attached the log for additional information. 

[stalling_playback.log](https://github.com/google/shaka-player/files/4578913/stalling_playback.log)

The MSe demo code which works fine on our platform:
```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>MSE Demo</title>
</head>
<body>
  <h1>MSE Demo</h1>
  <div>
    <video autoplay controls width="80%"></video>
  </div>

  <script type="text/javascript">
    (function() {
      var baseUrl = 'https://bitdash-a.akamaihd.net/content/MI201109210084_1/video/720_2400000/dash/';
      var initUrl = baseUrl + 'init.mp4';
      var templateUrl = baseUrl + 'segment_$Number$.m4s';
      var sourceBuffer;
      var index = 0;
      var numberOfChunks = 52;
      var video = document.querySelector('video');
        video.play();
      if (!window.MediaSource) {
        console.error('No Media Source API available');
        return;
      }

      var ms = new MediaSource();
      video.src = window.URL.createObjectURL(ms);
      ms.addEventListener('sourceopen', onMediaSourceOpen);

      function onMediaSourceOpen() {
        sourceBuffer = ms.addSourceBuffer('video/mp4; codecs="avc1.4d401f"');
        sourceBuffer.addEventListener('updateend', nextSegment);

        GET(initUrl, appendToBuffer);

        video.play();
      }

      function nextSegment() {
        var url = templateUrl.replace('$Number$', index);
        GET(url, appendToBuffer);
        index++;
        if (index > numberOfChunks) {
          sourceBuffer.removeEventListener('updateend', nextSegment);
        }
      }

      function appendToBuffer(videoChunk) {
        if (videoChunk) {
          sourceBuffer.appendBuffer(new Uint8Array(videoChunk));
        }
      }

      function GET(url, callback) {
        var xhr = new XMLHttpRequest();
        xhr.open('GET', url);
        xhr.responseType = 'arraybuffer';

        xhr.onload = function(e) {
          if (xhr.status != 200) {
            console.warn('Unexpected status code ' + xhr.status + ' for ' + url);
            return false;
          }
          callback(xhr.response);
        };

        xhr.send();
      }
    })();
  </script>
</body>
</html>
```

**Comment by joeyparrish** [comment]
We stopped supporting Safari 9 in commit 65370b4b994a3b11a9faa95ddda6f261c294d3d3, first released in v2.5.6 in November 2019.

We make our support decisions based on the browser share info at https://caniuse.com/usage-table .  When a browser or browser version's share drops below 1%, we may choose to end support for that browser or version.  Safari 9.1 is currently at 0.05% based on the caniuse.com data.

We used to have code that would effectively blacklist older Safari versions, but that was removed on the assumption that nobody was using them.  But perhaps that was premature.

Does this help?

**Comment by joeyparrish** [comment]
> We used to have code that would effectively blacklist older Safari versions, but that was removed on the assumption that nobody was using them. But perhaps that was premature.

I was mistaken.  We removed the MSE-based blacklisting of older Safari versions in v2.5.6, but we gained a new, higher-level blacklist in `shaka.Player.isBrowserSupported()` in v2.5.9 which rejects all versions of Safari < 12.

So you should get `false` from `shaka.Player.isBrowserSupported()`.  If you aren't calling it in your application, you should be.  I hope this helps!

Our README.md is out-of-date with respect to the minimum supported version of Safari (which it says is 11, not 12).  I'll update that shortly.

**Comment by harishnb** [comment]
sad to know that safari 9.1 is not supported unfortunately we cannot upgrade our browser in our embedded devices. We do get browser supported as false but the video plays for sometime before stalling. Is there any sequence diagram I can refer to know what exactly is happening? I saw the high level architecture but sequence diagram would be of great help. Once again thank you for prompt response.

**Comment by joeyparrish** [comment]
The problem is that WebKit had MediaSource-related bugs around Safari 9.  In fact, they still have MSE bugs in Safari 13.  We used to polyfill various versions of WebKit's MediaSource implementation to work around these, but the workarounds for Safari 9 and 10 have been removed from Shaka Player.  We are currently polyfilling Safari 11, 12, and 13 and officially supporting Safari 12 & 13.  Older versions of Shaka Player (v2.4, for example) should still support older versions of Safari, but we are no longer updating those older release branches.
